src/common/snippets/src/lowered/loop_manager.cpp

```diff
@@ -256,9 +256,11 @@ size_t LoopManager::replace_with_new_loop(const LinearIR& linear_ir,
                                           const size_t old_id) {
     const auto is_bound_explicit_loop_begin = ov::is_type<op::LoopBegin>(loop_begin_pos->get()->get_node());
     const auto is_bound_explicit_loop_end = ov::is_type<op::LoopEnd>(std::prev(loop_end_pos)->get()->get_node());
-    OPENVINO_ASSERT((is_bound_explicit_loop_begin && is_bound_explicit_loop_end) ||
-                        (!is_bound_explicit_loop_begin && !is_bound_explicit_loop_end),
-                    "Incorrect LoopBounds!");
+    const bool both_explicit = is_bound_explicit_loop_begin && is_bound_explicit_loop_end;
+    const bool both_implicit = !is_bound_explicit_loop_begin && !is_bound_explicit_loop_end;
+    if (!both_explicit) {
+        OPENVINO_ASSERT(both_implicit, "Incorrect LoopBounds!");
```

Comment:
So it does not allow to write assert(exp || impl)?

Reply:
No, it does not allow this